src/grt/src/fastroute/src/utility.cpp

```diff
@@ -278,18 +284,26 @@ void FastRouteCore::fillVIA()
 
         if (node2_alias < num_terminals && treenodes[node2_alias].hID == BIG_INT
             && edgeID == treenodes[node2_alias].lID) {
-          const int n2a_access_layer = nets_[netID]->getPinL()[node2_alias];
-
-          // Connect edge to pin n2 if not on the same layer
-          if (tmpL[newCNT - 1] != n2a_access_layer) {
-            int diff = n2a_access_layer - tmpL[newCNT - 1];
-            for (int i = 1; i <= abs(diff); i++) {
-              tmpX[newCNT] = tmpX[newCNT - 1];
-              tmpY[newCNT] = tmpY[newCNT - 1];
-              tmpL[newCNT] = tmpL[newCNT - 1] + (diff / abs(diff));
-              newCNT++;
-              numVIAT1++;
-            }
+          int pin_botL, pin_topL;
+          int edge_init = tmpL[newCNT - 1];
```

Comment:
@arthurjolo I see some errors in the regression tests that might be a crash. Perhaps it is related to this clang-tidy warning?

Reply:
this warning is actually weird, because tmpL[newCNT - 1] should always be populated. But still I changed how we use this variable, and changed other variables from _int_ to _short_ so we don't have convertion errors.